modules/backend/controllers/Files.php

```diff
@@ -77,6 +77,9 @@ protected static function getTemporaryUrl($file, $path = null)
             if (empty($path)) {
                 $path = $file->getDiskPath();
             }
+            if (!$disk->exists($path)) {
+                return '';
```

Comment:
Don't return an empty string here, null is the value that is to be returned when a valid URL can not be generated. Instead, wrap the remaining code ($expires and $url = Cache::remember) in `if ($disk->exists($path))` and add a comment explaining why the check is required (i.e. `// Some storage drivers will return a valid temporary URL even if the file does not exist`)

Reply:
Please review the new version